documentation/man/features/kw-signature.rst

```diff
@@ -0,0 +1,94 @@
+===================================================
+kw-signature - Write commit and patch trailer lines
+===================================================
+
+.. _signature-doc:
+
+SYNOPSIS
+========
+| *kw* *signature* (-s(<string>) | \--add-signed-off-by=(<string>)) [\--verbose] [<sha> | <patch>]
+| *kw* *signature* (-r(<string>) | \--add-reviewed-by=(<string>)) [\--verbose] [<sha> | <patch>]
+| *kw* *signature* (-a(<string>) | \--add-acked-by=(<string>)) [\--verbose] [<sha> | <patch>]
+| *kw* *signature* (-f(<sha>) | \--add-fixes=(<sha>)) [\--verbose] [<sha> | <patch>]
+
+DESCRIPTION
+===========
+This is a wrapper to some useful usage of the 'git commit --amend',
+'git rebase' and the 'git interpret-trailers' commands. This kw command
+is able to perform usual operations over trailer lines of commits and patches.
+By default, it uses the commit currently pointed by ``HEAD`` as the target
+of the operation if user does not specify one. Also, no operation is used by
+default. At least one operation option must be given.
+
+This command does not allow repetition of options. For instance, the
+following command will give an error::
+
+  $ kw signature -r"Reviewer Name <mail>" -r"Some Name <mail>"
```

Comment:
What if I want to add Reviewed-by from different developers?
For example, I got a Reviewed-by from Joe and another one from Bob, but for some reason, I have to do it manually; why not allow:

```bash
kw signature -r "Joe Doe <somethingxpto@lala.com" -r "Bob Lala <abcd@1234s.com"
```

This scenario looks valid to me. I think you can check for name/email duplication and avoid this case:

```bash
kw signature -r "Joe Doe <somethingxpto@lala.com"  -r "Joe Doe <somethingxpto@lala.com"
```

You can just add a warning in the above case.

Reply:
I've updated the code to allow repetition of options. Though I'm not checking name/email duplication yet.
By default the following command:

```sh
kw signature -r "Joe Doe <somethingxpto@lala.com>"  -r "Joe Doe <somethingxpto@lala.com>"
```

It only writes `Reviewed-by: Joe Doe <somethingxpto@lala.com>` once because it seems that both `git amend --trailer` and `git-interpret-trailers` will not write a duplicate line if the last trailer line is equals to the line it's trying to write right now.

Though it's not hard to come up with ways the user can still somehow force a duplicate line. For example:

```sh
kw signature -r -s -r -s
```

The above command will write:

```
Reviewed-by: Name <Mail>
Signed-off-by: Name <Mail>
Reviewed-by: Name <Mail>
Signed-off-by: Name <Mail>
```

I'm still thinking of ways to solve this issue. Maybe I'll have to, in fact, check all names/emails from each option and check if there are repeating options with the same signature.

Reply:
> I've updated the code to allow repetition of options. Though I'm not checking name/email duplication yet. By default the following command:
> 
> ```shell
> kw signature -r "Joe Doe <somethingxpto@lala.com>"  -r "Joe Doe <somethingxpto@lala.com>"
> ```

I suppose the second Joe was a duplication, right? I'll assume you meant Bob.

> 

Now that I'm rereading my comment and checking your reply, I think that a better interface would be:

```sh
kw signature -r "Joe Doe <somethingxpto@lala.com>, Bob Lala <abcd@1234s.com"
```

In other words, just use commas to separate different names.

> It only writes `Reviewed-by: Joe Doe <somethingxpto@lala.com>` once because it seems that both `git amend --trailer` and `git-interpret-trailers` will not write a duplicate line if the last trailer line is equals to the line it's trying to write right now.
> 

I guess this is what we want. However, you can check this yourself in the code and send a warning to the user.

> Though it's not hard to come up with ways the user can still somehow force a duplicate line. For example:
> 
> ```shell
> kw signature -r -s -r -s
> ```
> 

I'm sorry. I guess I was not clear in my original comments. In summary, we don't want duplication of tags with the same name and email. If the tags are different, it is okay to have them, but if they are equal, we don't want them.

> The above command will write:
> 
> ```
> Reviewed-by: Name <Mail>
> Signed-off-by: Name <Mail>
> Reviewed-by: Name <Mail>
> Signed-off-by: Name <Mail>
> ```

From your example, we don't want the duplicated Reviewed-by and Signed-off-by, we just want:

```
Reviewed-by: Name <Mail>
Signed-off-by: Name <Mail>
```